sqllogictest: start testing parser

benesch opened the pull request:
This has already shook out a few bugs.

jamii commented on src/sqllogictest/src/parser.rs:
```diff
@@ -94,28 +94,22 @@ pub fn parse_record<'a>(
         }
 
-        "halt" => Ok(Some(Record::Halt)),
+        "halt" => Ok(None),
```
This looks like you just skip halts now?

jamii replied:
Oh, no, it's the opposite, we halt on "substest" etc below which we used to just ignore.

benesch commented:
Dropped the parser changes in this PR and rebased on your fixes, Jamie, and happily the tests all apply to your changes as well. I pushed up another commit that turns panics into graceful errors that allow other files to continue. Going to merge so I can unblock getting stats on a full SLT run.
